test/stmt/foreach.swift

```diff
@@ -15,7 +15,7 @@ struct BadContainer2 : Sequence { // expected-error{{type 'BadContainer2' does n
 
 func bad_containers_2(bc: BadContainer2) {
   for e in bc { }
-  // expected-error@-1{{variable 'e' is not bound by any pattern}}
+  // expected-warning@-1 {{immutable value 'e' was never used; consider replacing with '_' or removing it}}
```

Comment:
Why did this and a few others below suddenly become valid?

Reply:
This is because `BadContainer2` is declared to be conforming to `Sequence`, although it's invalid conformance constraint succeeds. It used to fail on the dependent member type associated with sequence element but I made it so that is handled as a hole.

Reply:
Everywhere `variable 'e' is not bound by any pattern` exists on a non-recursive decl is a sign that we are either not invalidating the AST properly, or we aren't constructing constraint systems with sufficient information to invalidate an associated pattern binding specifically.  It's good to see movement here.